document/document.go

```diff
@@ -15,6 +16,17 @@ const (
 	ExpiresAtField = "_expiresAt"
 )
 
+var (
+	DocumentType       = reflect.ValueOf(&Document{}).Type()
+	DocumentObjectType = reflect.ValueOf(DocumentObject{}).Type()
+)
+
+// DocumentObject is another form of document, You can assign its Data to your structure
```

Comment:
Why do we need this structure?

Reply:
save() not only saves a document, it can also be used to update an existing document. When performing an update operation, the _id field is essential

Reply:
When take an parameter of any type instead of a Document object. If we want this to update an existing document, the primary key is required

Reply:
Yes, but any custom struct can specify an _id field using struct tags

Reply:
oh i didn't notice that